keanu-project/src/test/java/io/improbable/keanu/algorithms/mcmc/proposal/MultivariateGaussianProposalDistributionTest.java

```diff
@@ -0,0 +1,125 @@
+package io.improbable.keanu.algorithms.mcmc.proposal;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.improbable.keanu.KeanuRandom;
+import io.improbable.keanu.algorithms.Variable;
+import io.improbable.keanu.distributions.continuous.MultivariateGaussian;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+import io.improbable.keanu.vertices.intgr.probabilistic.PoissonVertex;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.closeTo;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class MultivariateGaussianProposalDistributionTest {
+
+    public Proposal proposal;
+    private static final DoubleTensor currentStateForVertex1 = DoubleTensor.create(4.2, 4.7);
+    private static final DoubleTensor currentStateForVertex2 = DoubleTensor.create(42.0, 48.0);
+
+    private static final DoubleTensor proposedStateForVertex1 = DoubleTensor.create(4.3, 5.8);
+    private static final DoubleTensor proposedStateForVertex2 = DoubleTensor.create(43.0, 32.0);
+
+    private static final DoubleTensor sigmaForVertex1 = DoubleTensor.create(1., 3.);
+    private static final DoubleTensor sigmaForVertex2 = DoubleTensor.create(2., 2.);
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Mock
+    private GaussianVertex vertex1;
+    @Mock
+    private GaussianVertex vertex2;
+
+    private MultivariateGaussianProposalDistribution proposalDistribution;
+    private Map<Variable, DoubleTensor> sigmas;
+
+    @Before
+    public void setUpProposalDistribution() {
+        sigmas = ImmutableMap.of(
+            vertex1, sigmaForVertex1,
+            vertex2, sigmaForVertex2
+        );
+        proposalDistribution = new MultivariateGaussianProposalDistribution(sigmas);
+    }
+
+    @Before
+    public void setRandomSeed() {
+        KeanuRandom.setDefaultRandomSeed(0);
+    }
+
+    @Before
+    public void setUpProposal() {
+        when(vertex1.getValue()).thenReturn(currentStateForVertex1);
```

Comment:
Lovely mocking but do you need it? What's wrong with just creating an actual GaussianVertex?

Reply:
Copied it from `GaussianProposalDistribution`

Reply:
I assumed the mock was to make it distribution agnostic?